doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,736 @@
+    - [Test case #1](#test-case-1)
+    - [Test case #2](#test-case-2)
+    - [Test case #3](#test-case-3)
+    - [Test case #4](#test-case-4)
+    - [Test case #5](#test-case-5)
+    - [Test case #6](#test-case-6)
+    - [Test case #7](#test-case-7)
+    - [Test case #8](#test-case-8)
+    - [Test case #9](#test-case-9)
+    - [Test case #10](#test-case-10)
+    - [Test case #11](#test-case-11)
+    - [Test case #12](#test-case-12)
+    - [Test case #13](#test-case-13)
+    - [Test case #14](#test-case-14)
+    - [Test case #15](#test-case-15)
+    - [Test case #16](#test-case-16)
+
+#### Overview
+The purpose is to test "RX_DRP" counter got from show command "show interfaces counters" triggers on receiving specific packets by DUT.
+The "RX_DRP" counter counts all discard events. This counter counts concurrently with other discard counters.
+The test assumes all necessary configuration are already pre-configured on the SONIC switch before test runs.
+Destination IP address of the injected packet must be routable to ensure packet should not be routed but dropped.
+
+#### Scope
+The purpose of test cases is to verify that "RX_DRP" counter triggers on SONIC system, making sure that specific traffic drops correctly, according to sent packet and configured packet discards.
+
+
+#### Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Discard groups covered by test cases
+Please refer to the test case for detailed description.
+
+| Test case ID| Drop reason | Group type|
+|-------------|-------------|-----------|
+| 1 | SMAC and DMAC are equal |Ethernet |
+| 2 | Not allowed VLAN TAG| Ethernet|
+| 3 | Multicast SMAC | Ethernet|
+| 4 | Reserved DMAC | Ethernet|
+| 5 | Loop-back filter | IP|
+| 6 | Packet exceed router interface MTU| IP|
+| 7 | Time To Live (TTL) Expired | IP|
+| 8 | Discard at router interface for non-routable packets | IP|
+| 9 | Absent IP header | IP|
+| 10 | Broken IP header - header checksum or IPver or IPv4 IHL too short | IP|
+| 11 | Unicast IP with multicast DMAC or broadcast DST MAC | IP|
+| 12 | DST IP is loopback address | IP|
+| 13 | SRC IP is loopback address | IP|
+| 14 | SRC IP is multicast address | IP|
+| 15 | SRC IP address is in class E | IP|
+| 16 | SRC IP address is not specified | IP|
+| 17 | DST IP address is not specified | IP|
+| 18 | ACL SRC IP DROP| IP|
+
+
+#### Related DUT CLI commands
+| **Command**                                                      | **Comment** |
+|------------------------------------------------------------------|-------------|
+| show interfaces counters              | Check ```RX_DRP```                     |
+| counterpoll rif enable                | Enable RIF counters                    |
+| show interface counters rif           | Show RIF counters                      |
+| aclshow -a                            | Check ```PACKETS COUNT```              |
+
+#### SAI attributes
+```SAI_PORT_STAT_IF_IN_DISCARDS``` - query number of all discards
+
+#### General test flow
```

Comment:
i would specify which step like
step #1: stop vms . Note: Didnt nazarii said how and not not stop but disable?
step #2: run test case ...
step #3: start vms . Note: as Nazarri commented

Reply:
Added steps.